sdk/core/core/src/az_http_request.c

```diff
@@ -147,6 +147,11 @@ AZ_NODISCARD az_result
 az_http_request_append_header(_az_http_request* p_request, az_span key, az_span value)
 {
   AZ_PRECONDITION_NOT_NULL(p_request);
+
+  // remove white spaces from kay and value
+  key = _az_span_trim_white_space(key);
```

Comment:
Do we want to trim whitespace from both beginning and end for the key? Is it possible for the key to have leading whitespace?

Reply:
Yes, trim both sides